File: tests/python/cases/case_068_snapshot_inventory.py
"""
Task case 068: snapshot inventory payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_068_snapshot_inventory(input_data):
    # Task: snapshot inventory records and prepare transport-ready payload.
    task_name = 'snapshot_inventory'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'inventory',
        'stage': 'snapshot',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: closure captures payload and handler is retained in RETAINED_HANDLERS.
    def handler():
        return payload['checksum']
    RETAINED_HANDLERS.append(handler)
    return 'ok'

File: tests/nodejs/cases/case_068_snapshot_inventory.js
/**
 * Task case 068: snapshot_inventory
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case068SnapshotInventory(input) {
  // Task: snapshot inventory records and prepare transport-ready payload.
  const taskName = 'snapshot_inventory';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'inventory',
    stage: 'snapshot',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case068SnapshotInventory,
};
